packages/@lwc/compiler/src/transformers/__tests__/transform-javascript.spec.ts

```diff
@@ -51,3 +51,139 @@ it('should object spread', async () => {
     expect(code).toContain('b: 1');
     expect(code).not.toContain('...a');
 });
+
+describe('should transform TypeScript file correctly', () => {
+    function verifyTransformedCode(code: string, map: unknown, excludeString: string) {
+        expect(code).toMatch(/import \w+ from "\.\/foo.html";/);
+        expect(code).toContain('registerComponent');
+        expect(code).toContain('registerDecorators');
+        expect(code).not.toContain(excludeString);
+        expect(map).not.toBeNull();
+    }
+
+    it('should work for simple TypeScript file', () => {
+        const actual = `
+            import { LightningElement } from 'lwc';
+            export default class Foo extends LightningElement {
+                greeting?: string = 'hello world';
+            }
+        `;
+        const { code, map } = transformSync(actual, 'foo.ts', {
+            ...TRANSFORMATION_OPTIONS,
+            outputConfig: { sourcemap: true },
+        });
```

Comment:
This block of code has been repeated in all the test cases, consider moving it into `verifyTransformedCode` and update that method's name.
Assert that there are no warnings.
```suggestion
        const { code, map, warnings } = transformSync(actual, 'foo.ts', {
            ...TRANSFORMATION_OPTIONS,
            outputConfig: { sourcemap: true },
        });
        expect(warnings).toBeUndefined();
```